src/cascadia/TerminalSettingsEditor/EditAction.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "EditAction.h"
+#include "EditAction.g.cpp"
+#include "LibraryResources.h"
+#include "../TerminalSettingsModel/AllShortcutActions.h"
+
+using namespace winrt::Windows::UI::Xaml;
+using namespace winrt::Windows::UI::Xaml::Navigation;
+
+namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+{
+    EditAction::EditAction()
+    {
+        InitializeComponent();
+    }
+
+    void EditAction::OnNavigatedTo(const NavigationEventArgs& e)
+    {
+        const auto args = e.Parameter().as<Editor::NavigateToCommandArgs>();
+        _ViewModel = args.Command();
+        _ViewModel.PropagateWindowRootRequested([windowRoot = args.WindowRoot()](const IInspectable& /*sender*/, const Editor::ArgWrapper& wrapper) {

```

Comment:
do we ever unregister these? or do we hook up more and more and more if you navigate out of and back into a Command?